diff --git a/Makefile b/Makefile
index 52a4010..918f55b 100644
--- a/Makefile
+++ b/Makefile
@@ -469,6 +469,20 @@ initial_content:
 	curl -u admin:$(shell cat secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD) -H "Content-Type: application/json" -d "@demo-data/homepage.json" https://${DOMAIN}/node?_format=json
 	curl -u admin:$(shell cat secrets/live/DRUPAL_DEFAULT_ACCOUNT_PASSWORD) -H "Content-Type: application/json" -d "@demo-data/browse-collections.json" https://${DOMAIN}/node?_format=json
 
+.PHONY: deploy-federated-search
+.SILENT: deploy-federated-search
+deploy-federated-search: 
+
+	cd $(CURDIR)
+	rm -rf  islandora_lite_installation
+	git clone https://github.com/digitalutsc/islandora_lite_installation
+	rm -rf codebase/islandora_lite_installation
+	mv islandora_lite_installation codebase/islandora_lite_installation
+
+	chmod +x codebase/islandora_lite_installation/scripts/*.*
+
+	docker-compose exec -T drupal with-contenv bash -lc "islandora_lite_installation/scriptsfederated_search.sh"
+
 .PHONY: clean
 .SILENT: clean
 ## Destroys everything beware!
