diff --git a/src/IslandoraBreadcrumbBuilder.php b/src/IslandoraBreadcrumbBuilder.php
index ae13aad..d550050 100644
--- a/src/IslandoraBreadcrumbBuilder.php
+++ b/src/IslandoraBreadcrumbBuilder.php
@@ -252,17 +252,15 @@ class IslandoraBreadcrumbBuilder implements BreadcrumbBuilderInterface {
    */
   protected function getViewLink(Node $node) {
     $nid = $node->id();
-    if (Term::load($node->get('field_model')->target_id)->get('name')->value === "Collection") {
+    /*if (Term::load($node->get('field_model')->target_id)->get('name')->value === "Collection") {
       $url_object = \Drupal::service('path.validator')->getUrlIfValid("/collection/%node");
       if ($url_object == FALSE)
         return $node->toLink();
       else
         $route_name = $url_object->getRouteName();
-      /* If the parent is collection,
-      replace the node link with collection view link. */
       return Link::createFromRoute($node->getTitle(), $route_name, ['node' => $nid]);
-    }
-    elseif (Term::load($node->get('field_model')->target_id)->get('name')->value === "Paged Content") {
+    }*/
+    if (Term::load($node->get('field_model')->target_id)->get('name')->value === "Paged Content") {
       return Link::createFromRoute($node->getTitle(), "entity.node.canonical", ['node' => $node->id()]);
     }
     else {
