diff --git a/js/leaflet_geojson.bbox.js b/js/leaflet_geojson.bbox.js
index 30a6eb8..e8823d3 100644
--- a/js/leaflet_geojson.bbox.js
+++ b/js/leaflet_geojson.bbox.js
@@ -1,4 +1,4 @@
-(function ($, Drupal, drupalSettings) {
+(function (Drupal, $, once, drupalSettings) {
   'use strict';
 
   var jsonrequest;
@@ -8,9 +8,34 @@
     markerGroup: null,
     overlays: {},
 
-    onMapLoad: function (map, settings) {
+    onMapLoad: function (map) {
       Drupal.leafletBBox.map = map;
       Drupal.leafletBBox.markerGroup = [];
+      var resetControl = L.control({position: 'topleft'});
+      
+      map.eachLayer(function (layer) {
+        if (layer instanceof L.TileLayer || layer instanceof L.ImageOverlay) {
+          layer.options.noWrap = true;
+        }
+      });
+
+      resetControl.onAdd = function (map) {
+        var controlDiv = L.DomUtil.create('div', 'leaflet-control-resetview leaflet-bar leaflet-control');
+        var link = L.DomUtil.create("a", "leaflet-bar-part leaflet-bar-part-single", controlDiv);
+        link.title = "Reset View";
+        link.href = "#";
+        link.setAttribute("role", "button");
+        var icon = L.DomUtil.create("span", "leaflet-control-resetview-icon", link);
+        icon.innerHTML = `<div style="font-size: 1.5rem;">↻</div>`;
+
+        link.onclick = function () {
+          map.setView([0, 0], 2);
+        }
+
+        return controlDiv;
+      };
+
+      resetControl.addTo(map);
 
       // Intialize empty layers and associated controls.
       var layer_count = 0;
@@ -35,8 +60,8 @@
       }
 
       // settings maxbound so that the map remain alwxays centered
-      var southWest = new L.LatLng(90, -18000);
-      var northEast = new L.LatLng(-90, 180000);
+      var southWest = new L.LatLng(90, -180);
+      var northEast = new L.LatLng(-90, 180);
       var bounds = new L.LatLngBounds(southWest, northEast);
 
       map.setMaxBounds(bounds);
@@ -236,10 +261,10 @@
   };
 
   // Insert map.
-  $(document).bind('leaflet.map', function (e, settings, lMap) {
-    $(window, settings).once("leaflet_geojson_bbox_init_map").each( function(){
-      Drupal.leafletBBox.onMapLoad(lMap, settings);
+  $(document).bind('leaflet.map', function (e, map, lMap) {
+    $(once('leaflet_geojson_bbox_init_map', 'html')).each(function() {
+      Drupal.leafletBBox.onMapLoad(lMap);
     });
   });
 
-})(jQuery, Drupal, drupalSettings);
+})(Drupal, jQuery, once, drupalSettings);
