diff --git a/src/Plugin/Field/FieldFormatter/PdfDefault.php b/src/Plugin/Field/FieldFormatter/PdfDefault.php
old mode 100755
new mode 100644
index 6d1f851..7bf1fb8
--- a/src/Plugin/Field/FieldFormatter/PdfDefault.php
+++ b/src/Plugin/Field/FieldFormatter/PdfDefault.php
@@ -168,6 +168,9 @@ class PdfDefault extends FormatterBase {
       if ($item->entity->getMimeType() == 'application/pdf') {
         $file_url = \Drupal::service('file_url_generator')->generateAbsoluteString($item->entity->getFileUri());
         $iframe_src = \Drupal::service('file_url_generator')->generateAbsoluteString($viewer_path) . '?file=' . rawurlencode($file_url);
+        if (strpos($iframe_src, "http") !== false) {
+            $iframe_src = str_replace("http:", "https:", $iframe_src);
+        }
         $iframe_src = !empty($query) && $keep_pdfjs ? $iframe_src . '#' . $query : $iframe_src;
         $html = [
           '#theme' => 'file_pdf',
